sgx-types/src/ssa.rs

```diff
@@ -137,32 +200,48 @@ bitflags! {
 #[derive(Debug)]
 #[repr(C)]
 pub struct ExceptionInfo {
+    /// In case of a page fault, contains the linear address that caused the fault.
     pub maddr: u64,
+    /// Exception error code for GP fault or page fault.
+    /// TODO: Should this have more options in case of general purpose fault?
```

Comment:
Do you want to resolve this TODO in this PR?

Reply:
I'm happy to do it, but I wanted Nathaniel's thoughts on it in case that idea doesn't make sense with his overall structure.

Reply:
I'm not sure I understand the question.

Reply:
It's possible that I'm misunderstanding the code here. But In the SGX documentation, the `errcd` is used both for GP faults and page faults. In our code, the `errcd` always holds a `PageFault`. Is this deliberate? I can't recall the reasoning behind this or if it matters, or if this is taken care of elsewhere.

Reply:
I honestly don't remember. Please file a separate issue for this. Please make it a priority. The code I'll write next week will probably depend on this struct.

Reply:
Sure - I just added #135 and I'll look into this. I don't yet have a great grasp of how these errors get populated into the Rust structs, but I can at least try to set up a complementary GP struct for `errcd`.